rmgpy/rmg/main.py

```diff
@@ -1463,21 +1481,126 @@ def generateCanteraFiles(self, chemkinFile, **kwargs):
+                            trimolecularThreshold_restart = f.get('trimolecularThreshold').value
+
+                        # Expand Thresholds to match number of species in the current model.
+                        unimolecularThreshold = np.zeros(numCoreSpecies, bool)
+                        unimolecularThreshold[:numRestartSpcs] = unimolecularThreshold_restart
+
+                        bimolecularThreshold = np.zeros((numCoreSpecies, numCoreSpecies), bool)
+                        bimolecularThreshold[:numRestartSpcs, :numRestartSpcs] = bimolecularThreshold_restart
+
+                        if self.trimolecular:
+                            trimolecularThreshold = np.zeros((numCoreSpecies, numCoreSpecies, numCoreSpecies), bool)
+                            trimolecularThreshold[:numRestartSpcs, :numRestartSpcs, :numRestartSpcs] = \
+                                trimolecularThreshold_restart
+
+                        filters_found = True
+
+                    except KeyError:  # If we can't find the filters then this is because filtering was not used
+                        logging.warning('No filters were found in file {0}. This is to be expected if the restart '
+                                        'files specified came from an RMG job that did not use filtering. Therefore, '
+                                        'RMG will assume that all of the core species from the restart seed have '
+                                        'already been reacted and will not react them again. Additional species that '
+                                        'were added to the input file but are not in the restart core seed WILL be '
+                                        'reacted with the rest of the core though. If this is not the intended '
+                                        'behavior please stop this job and correct the missing '
+                                        'filters.'.format(self.filtersPath))
+
+                        filters_found = False
+
+                # Reorder the core species to match the indices of the restart filter tensors
+                core_species_map = {}
+
+                for index, spcs in restartSpeciesMap.items():
+                    for j, core_spcs in enumerate(self.reactionModel.core.species):
+                        if core_spcs.isIsomorphic(spcs):
+                            core_species_map[index] = core_spcs
+                            break
+                    else:
+                        raise RuntimeError('Species {0} was defined in the restart file, but was not included in the'
+                                           'core.'.format(spcs))
+                    self.reactionModel.core.species.pop(j)
+
+                remaining_species = self.reactionModel.core.species
+
+                reordered_core_species = [core_species_map[i] for i in range(len(core_species_map))] + remaining_species
+                self.reactionModel.core.species = reordered_core_species
```

Comment:
I think the core species map also doesn't need to be a dictionary.
```
reordered_core_species = []
for spc in restart_species_list:  # if it were a list instead of a dict
    for j, core_spcs in enumerate(self.reactionModel.core.species):
        if core_spcs.isIsomorphic(spc):
            reordered_core_species.append(self.reactionModel.core.species.pop(j)
            break
    else:
        raise RuntimeError

self.reactionModel.core.species = reordered_core_species + self.reactionModel.core.species
```

Reply:
Done